src/searchModal.js

```diff
@@ -98,16 +119,106 @@ export default function searchModalFactory(config) {
             .focus();
     }
 
+    /**
+     * Inits class filter selector
+     */
+    function initClassFilter() {
+        return new Promise(function (resolve) {
+            const rootClassUri = instance.config.rootClassUri;
+            const initialClassUri =
+                instance.config.criterias && instance.config.criterias.class
+                    ? instance.config.criterias.class
+                    : instance.config.rootClassUri;
+            resourceSelector = resourceSelectorFactory($('.class-tree', instance.getElement()), {
+                //set up the inner resource selector
+                selectionMode: 'single',
+                selectClass: true,
+                classUri: rootClassUri,
+                showContext: false,
+                showSelection: false
+            });
+
+            // when a class query is triggered, update selector options with received resources
+            resourceSelector.on('query', function (params) {
+                if (instance.config.testMocks.classTree) {
+                    resourceSelector.update(instance.config.testMocks.classTree, params);
+                    return;
+                }
```

Comment:
Please avoid using such a hardcoded bypass. This is introducing unnecessary complexity, and this is error prone.

Code must be testable indeed, but it doesn't need to explicitly support it. Eventually, you can add events or entry points to ease it, but you must not add a mode for this purpose. This is breaking the reliability of the test.

```suggestion
```